app/Http/Controllers/MaterialController.php

```diff
@@ -40,6 +42,52 @@ public function index()
     }
 
     /**
+     * Show the form for editing the specified resource.
+     *
+     * @param int $id of material
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $material = $this->material->findOrFail($id);
```

Comment:
I think I don't need try catch for function findOrFail because have exception handel